libmscore/measurebase.cpp

```diff
@@ -369,6 +369,7 @@ bool MeasureBase::setProperty(Pid id, const QVariant& value)
                   break;
             }
       score()->setLayoutAll();
+      score()->setPlaylistDirty();
```

Comment:
As far as I understand this line should be the only change that is needed to fix the issue. In case of any edits to a score `setProperty()` should probably be called for the corresponding measures both in MasterScore and in excerpt scores so the override marking the repeat list obsolete should be called anyway. Or do other changes in this PR also make a difference?

Reply:
Well, this is the only change needed to fix it, yes. But, I couldn't see a reason for having a masterscore override - I think it's better to always make the repeat list dirty when setting the playlist dirty, no matter whether it's the method of score or masterscore being called. That's why I removed the override, so that set playlist dirty will always make the repeat list dirty. Thoughts?

Reply:
As I recall, it doesn't fix it for jumps without the other change, or making a change to jumps also

Reply:
Ah yes, that was it as well. Thanks for the reminder :)

Reply:
The reason why the current solution may be preferable is that it allows to avoid exposing extra details related to the repeat list to the interface of MasterScore class. It is a purely internal MasterScore implementation detail that RepeatList is stored within it and its state is somehow tracked (it could be generated each time on demand, for example), and in my opinion it would be better to avoid or at least minimize manipulating this detail from other classes. Of course, the current solution is not clean either, ideally I would move both RepeatList and that `_playlistDirty` flag to one class, either MasterScore or Score, which would allow not to have this function implemented both in Score and MasterScore. But I don't currently see a benefit of the solution proposed here over the current one.

It also doesn't seem probable that those changes fix something that is not fixed by this line alone. I believe the issues that were present with changing this line alone will still remain after applying the whole patch as it stands now. Of course, it would be great if you could provide an example of the opposite, but if my understanding is correct, those changes shoudn't make any difference regarding this issue.

Reply:
Well, then let's move _playlistDirty to masterscore and redirect score::playlistDirty and setplaylistdirty to masterscore. That way we can avoid exposing repeat list while still always updating it when needed.

Reply:
I've made the changes.

Reply:
@dmitrio95 Maybe I'm missing something, but this one line alone doesn't fix the problem at all, simply because the repeatlist never gets updated unless somehow the repeatlist gets updated, and currently, calling Score::setPlaylistDirty() just doesn't do it.  So right now, both repeat barlines *and* jumps are broken up, because jumps *only* call Score::setPlaylistDirty() which isn't enough, and repeat barlines do nothing.  This one change alone would make repeat barlines work the same as jumps - which is to say, both would still be broken, just more consistently so.

So, I say we need this line to put repeat barlines on the same footing as jumps, but we *also* need to somehow make sure that calling Score::setPlaylistDirty() gets the repeat lsit marked dirty as well.  I don't care much about which function expose what to whom, so long as one way or another, both MeasureBase::setProperty() and Jump::setProperty() get the job done.  Hmm, no idea if Marker needs to be involved also.

`Again, the thing that is *not* fixed by this line alone is jumps.  They suffer the same problem as repeat barlines in current master.  So I don't see how a change to MeasureBase alone could possibly help with that.  And again, without some additional change, calling Score::setPlaylistDirty(0 is not sufficient in eitherbecause masterscore repeatlist never gets updated.

Reply:
Bottom line for me, I *think* the PR as it stands at this moment is good, because it ticks the two boxes it needs to for me (calling Score::setPlaylistDirty() where it needs to, and making sure that in turns make the repeatlist dirty), but I'm not feeling in a position to really make the call regarding the class interface etc.  It does look liek this is a bigger change than it *needs* to be, but it also looks pretty clean to me.